rpc_test.go

```diff
@@ -88,3 +92,181 @@ func TestInboundsOrderAfterStart(t *testing.T) {
 	httpInbound := inbounds[1].(http.Inbound)
 	assert.NotNil(t, httpInbound.Addr(), "expected an HTTP addr")
 }
+
+func TestStartStopFailures(t *testing.T) {
+	tests := []struct {
+		desc string
+
+		inbounds  func(*gomock.Controller) []transport.Inbound
+		outbounds func(*gomock.Controller) transport.Outbounds
+
+		wantStartErr string
+		wantStopErr  string
+	}{
+		{
+			desc: "all success",
+			inbounds: func(mockCtrl *gomock.Controller) []transport.Inbound {
+				inbounds := make([]transport.Inbound, 10)
+				for i := range inbounds {
+					in := transporttest.NewMockInbound(mockCtrl)
+					in.EXPECT().Start(gomock.Any()).Return(nil)
+					in.EXPECT().Stop().Return(nil)
+					inbounds[i] = in
+				}
+				return inbounds
+			},
+			outbounds: func(mockCtrl *gomock.Controller) transport.Outbounds {
+				outbounds := make(transport.Outbounds)
+				for i := 0; i < 10; i++ {
+					out := transporttest.NewMockOutbound(mockCtrl)
+					out.EXPECT().Start().Return(nil)
+					out.EXPECT().Stop().Return(nil)
+					outbounds[fmt.Sprintf("service-%v", i)] = out
+				}
+				return outbounds
+			},
+		},
+		{
+			desc: "inbound 6 start failure",
+			inbounds: func(mockCtrl *gomock.Controller) []transport.Inbound {
+				inbounds := make([]transport.Inbound, 10)
+				for i := range inbounds {
+					in := transporttest.NewMockInbound(mockCtrl)
+					if i == 6 {
+						in.EXPECT().Start(gomock.Any()).Return(errors.New("great sadness"))
+					} else {
+						in.EXPECT().Start(gomock.Any()).Return(nil)
+					}
+					// no Stop() because Start() failed
+					inbounds[i] = in
+				}
+				return inbounds
+			},
+			outbounds: func(mockCtrl *gomock.Controller) transport.Outbounds {
+				outbounds := make(transport.Outbounds)
+				for i := 0; i < 10; i++ {
+					out := transporttest.NewMockOutbound(mockCtrl)
+					out.EXPECT().Start().Return(nil)
+					outbounds[fmt.Sprintf("service-%v", i)] = out
+				}
+				return outbounds
+			},
+			wantStartErr: "great sadness",
```

Comment:
No, this is an `errorGroup` containing a collection of all `Start()/Stop()` failures.